src/main/java/emissary/core/Stage.java

```diff
@@ -1,136 +1,240 @@
+        }
+
+        StageEnum(int index) {
+            this.index = index;
         }
-        return false;
     }
 
+    public StageEnum getStageEnum() {
+        return stageEnum;
+    }
 
-    /**
-     * Return stage name at index
-     */
-    public String getStageName(final int i) {
-        if (i >= 0 && i < stages.size()) {
-            return stages.get(i);
-        } else {
-            return "UNDEFINED";
-        }
+    public void setStageEnum(StageEnum stageEnum) {
+        this.stageEnum = stageEnum;
     }
 
-    /**
-     * Return stage index for name
-     * 
-     * @param stage the name of the stage
-     * @return the index of the stage or -1 if no such stage
-     */
-    public int getStageIndex(final String stage) {
-        for (int i = 0; i < stages.size(); i++) {
-            if (stages.get(i).equals(stage)) {
-                return i;
-            }
+    public boolean isParallelStage() {
+        return this.stageEnum.isParallelStage();
+    }
+
+    public String getStageName() {
+        return this.stageEnum.getStageName();
+    }
+
+    public StageEnum getStage(int index) {
+        switch (index) {
+            case 0:
+                return StageEnum.STUDY;
+            case 1:
+                return StageEnum.ID;
+            case 2:
+                return StageEnum.COORDINATE;
+            case 3:
+                return StageEnum.PRETRANSFORM;
+            case 4:
+                return StageEnum.TRANSFORM;
+            case 5:
+                return StageEnum.POSTTRANSFORM;
+            case 6:
+                return StageEnum.ANALYZE;
+            case 7:
+                return StageEnum.VERIFY;
+            case 8:
+                return StageEnum.IO;
+            case 9:
+                return StageEnum.REVIEW;
+            default:
+                return StageEnum.UNDEFINED;
         }
-        return -1;
     }
 
+    public StageEnum getStage(String name) {
+        switch (name) {
+            case "STUDY":
+                return StageEnum.STUDY;
+            case "ID":
+                return StageEnum.ID;
+            case "COORDINATE":
+                return StageEnum.COORDINATE;
+            case "PRETRANSFORM":
+                return StageEnum.PRETRANSFORM;
+            case "TRANSFORM":
+                return StageEnum.TRANSFORM;
+            case "POSTTRANSFORM":
+                return StageEnum.POSTTRANSFORM;
+            case "ANALYZE":
+                return StageEnum.ANALYZE;
+            case "VERIFY":
+                return StageEnum.VERIFY;
+            case "IO":
+                return StageEnum.IO;
+            case "REVIEW":
+                return StageEnum.REVIEW;
+            default:
+                return StageEnum.UNDEFINED;
+        }
+    }
 
-    /**
-     * Return a list of stages
-     */
     public List<String> getStages() {
-        return new ArrayList<String>(stages);
+        List<String> stages = new ArrayList<>();
```

Comment:
It feels like this list could be calculated in an init function and just returned when called since this list shouldn't change during runtime.